### PR TITLE
Optimize gulpfile.js

### PR DESCRIPTION
**Pull request by zenflow**
Changes
- Naming and organization
- Watches lintable files for linting, and buildable files for building (separate)
- In active development mode, doesn't wait for linting results before starting build.
- Doesn't repeat work, i.e. sources are compiled once, minified once, concatenated twice (.all.js & .all.min.js)
- Doesn't have two separate entry-points to _(edit) test_. We can remove the terrible hack in `.travis.yml`
- Uses minified css in sweetalert2.all.min.js  (see bundlesize report for byte savings) _(edit) and pretty css (as js) in sweetalert2.all.jss_
- Exposes direct control over when minification happens, with --skip-minification
- Exposes direct control over when the build includes .all.js files, with --skip-standalone

### CODE REVIEW

**Comment by zenflow** [comment]
Basically the rest of the changes I wanted to make in #826

**Comment by zenflow** [comment]
bundlesize only reports 585 bytes saved? that doesn't seem right..

**Comment by zenflow** [comment]
Forgot to mention a change
- cleaner way of setting `swal.version` in build step, source code includes `import {version} from '../package.json'`

File: package.json [hunk 1]
```diff
@@ -19,9 +19,13 @@
     "detect-browser": "^2.1.0",
     "event-stream": "^3.3.4",
     "execa": "^0.9.0",
-    "gulp": "^3.9.1",
+    "gulp": "github:gulpjs/gulp#4.0",
```
Comment (limonte) [inline package.json]: Can we somehow get rid of this change? I won't be able to use my globally installed gulp@3.9.1 for SweetAlert2,

Comment (zenflow) [reply package.json]: I was pretty sure that the globally installed gulp cli delegates to the project's locally installed gulp

Did you try it? If not I'll test it out

Comment (limonte) [reply package.json]: This is what's happening in my env when I try to run any gulp task:

~[](https://i.imgur.com/mLNLMFM.png)

Comment (zenflow) [reply package.json]: Ok. Why do you need to use your globally installed gulp?

Comment (limonte) [reply package.json]: btw. wtf is happening with gulp? why 4.0.0 isn't installable? https://github.com/gulpjs/gulp/releases/tag/v4.0.0

Comment (zenflow) [reply package.json]: We could add a package script `"gulp": "gulp"` so you can pass arguments in through `npm`, e.g. `npm run gulp -- build:styles`.

Comment (limonte) [reply package.json]: > Ok. Why do you need to use your globally installed gulp?

Good question :) I don't need it. Resolved.

**Comment by zenflow** [comment]
@limonte Glad you're satisfied with the byte savings! I actually expected it to be more ☹️

**Pull request merged**